lib/template/search-html.php

```diff
@@ -10,10 +10,60 @@
 
     <?php include(CONST_BasePath.'/lib/template/includes/html-top-navigation.php'); ?>
 
-    <form class="form-inline" role="search" accept-charset="UTF-8" action="<?php echo CONST_Website_BaseURL; ?>search.php">
-        <div class="form-group">
-            <input id="q" name="q" type="text" class="form-control input-sm" placeholder="Search" value="<?php echo htmlspecialchars($sQuery); ?>" >
+    <div class="top-bar" id="structured-query-selector">
+        <div class="search-type-link">
+            <a id="switch-to-reverse" href="<?php echo CONST_Website_BaseURL; ?>reverse.php?format=html">reverse search</a>
+        </div>
+        <?php
+        if (empty($aMoreParams['q']) and empty($aMoreParams['street']) and empty($aMoreParams['city']) and empty($aMoreParams['county']) and empty($aMoreParams['state']) and empty($aMoreParams['country']) and empty($aMoreParams['postalcode'])) {
+            echo '<div class="radio-inline"><input type="radio" name="query-selector" id="simple" value="simple" checked="checked">';
+            echo '<label for="simple">simple</label></div>';
+            echo '<div class="radio-inline"><input type="radio" name="query-selector" id="structured" value="structured">';
+            echo '<label for="structured">structured</label></div>';
+        }
+        elseif (!empty($aMoreParams['q'])) {
+            echo '<div class="radio-inline"><input type="radio" name="query-selector" id="simple" value="simple" checked="checked">';
+            echo '<label for="simple">simple</label></div>';
+            echo '<div class="radio-inline"><input type="radio" name="query-selector" id="structured" value="structured">';
+            echo '<label for="structured">structured</label></div>';
+        }
+        else {
+            echo '<div class="radio-inline"><input type="radio" name="query-selector" id="simple" value="simple">';
+            echo '<label for="simple">simple</label></div>';
+            echo '<div class="radio-inline"><input type="radio" name="query-selector" id="structured" value="structured" checked="checked">';
+            echo '<label for="structured">structured</label></div>';
+        }
+        ?>
+
+    <form role="search" accept-charset="UTF-8" action="<?php echo CONST_Website_BaseURL; ?>search.php">
+        <div class="form-group-simple"
+        <?php
+        if (empty($aMoreParams['q']) and empty($aMoreParams['street']) and empty($aMoreParams['city']) and empty($aMoreParams['county']) and empty($aMoreParams['state']) and empty($aMoreParams['country']) and empty($aMoreParams['postalcode'])) {
+            echo "style='display:block;'";
+        }
+        elseif (empty($aMoreParams['q'])) {
+            echo "style='display:none;'";
+        }
+        ?>>
+            <input id="q" name="q" type="text" class="form-control input-sm" placeholder="Search" value="<?php echo $aMoreParams['q']; ?>" >
```

Comment:
Whenever printing (echo) a value it should go through `htmlspecialchars()` function to avoid cross-side-scripting (somebody might search for something with `<` and break the HTML otherwise).